packages/next/src/executors/build/build.impl.ts

```diff
@@ -21,7 +26,32 @@ export default async function buildExecutor(
   context: ExecutorContext
 ) {
   const root = resolve(context.root, options.root);
-  const config = await prepareConfig(PHASE_PRODUCTION_BUILD, options, context);
+
+  const projGraph = createProjectGraph();
+  const { dependencies } = calculateProjectDependencies(
+    projGraph,
+    context.root,
+    context.projectName,
+    context.targetName,
+    context.configurationName
+  );
+  if (
+    !checkDependentProjectsHaveBeenBuilt(
+      context.root,
+      context.projectName,
+      context.targetName,
+      dependencies
+    )
+  ) {
+    throw new Error();
```

Comment:
do we want to use an error message here?

Reply:
Good point.. I'm looking at making this more straight-forward for folks using it. Will update the PR soon.